### ISSUE TITLE
1.8.1 release : Unable to send userauth-publickey request

### ISSUE THREAD

**Issue by jonathanturcotte** [libssh2/libssh2#336]
Our project uses libgit2 with libssh2 for cloning over SSH. After updating libssh2 to v1.8.1, libgit2 reports that it can no longer clone over SSH with the error : Unable to send userauth-publickey request. 

Dropping in libssh2 1.8.0 or a libssh2 built off master (commit 452517d96c2a636fa0e2f31cfa8b463886de7611) has it working fine again. My suspicion as a result is that the 1.8.1 release is bugged, but I'm not really certain how to debug it any further. 

I've confirmed that it happens on RHEL7, Ubuntu 16, and macOS.

**Comment by tcamin** [comment]
see https://www.libssh2.org/mail/libssh2-devel-archive-2019-03/0013.shtml

**Comment by jonathanturcotte** [comment]
Ah, that would do it. 

Is there a plan to release a 1.8.2 with that fix, or just should I just wait for 1.9.0?

**Comment by kdudka** [comment]
> Is there a plan to release a 1.8.2 with that fix, or just should I just wait for 1.9.0?

Not for me to decide but releasing 1.8.2 sounds like a good idea to me.  The 1.8.1 release went out badly broken -- even the embedded upstream smoke test did not pass.  There might be users that prefer to install security updates only and the 1.8.1 release would unnecessarily break their deployments.

**Comment by bagder** [comment]
Is the [1.8.1 branch](https://github.com/libssh2/libssh2/tree/1.8.1) working? /cc @willco007 

We could make a quick 1.8.2 release from that...

**Comment by kdudka** [comment]
As far as I can tell, d06ab72a146be1a5975d351e747cebe9f2b5357a works fine.

**Comment by kdudka** [comment]
By the way, the branch should be renamed to `1.8.x`.  Pushing additional commits to a branch named 1.8.1 when the 1.8.1 release is already out is confusing.

**Comment by bagder** [comment]
Agreed about the confusion. I created a [1.8.x](https://github.com/libssh2/libssh2/tree/1.8.x) branch just now to release a 1.8.2 from. Leaving the 1.8.1 as-is since it was already used and discussed for the 1.8.1 release.

**Comment by bagder** [comment]
1.8.2 has been released!

**Comment by n-rodriguez** [comment]
Hi there! Still not working here... :/

**Comment by tiennou** [comment]
Is that against 1.8.2 ? master ?

Can you share a few datapoints on the failing setup ? Is that against one of the big git providers ?

**Comment by n-rodriguez** [comment]
> Can you share a few datapoints on the failing setup ?

I run ArchLinux. Here libssh2 version:
```sh
nicolas@tchoum-laptop:~$ pacman -Qs libssh2
local/lib32-libssh2 1.8.2-1
    A library implementing the SSH2 protocol as defined by Internet Drafts (32-bit)
local/libssh2 1.8.2-1
    A library implementing the SSH2 protocol as defined by Internet Drafts
```

**Comment by n-rodriguez** [comment]
Here the script I use to test libssh2 with [rugged]() :+1: 

```ruby
#!/usr/bin/env ruby

begin
  require "bundler/inline"
rescue LoadError => e
  $stderr.puts "Bundler version 1.10 or later is required. Please update your Bundler"
  raise e
end

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem 'gitolite-rugged', github: 'jbox-web/gitolite-rugged', branch: 'devel'
end

require 'gitolite'
require 'yaml'

puts Rugged.features

# settings = { public_key: '/home/nicolas/.ssh/redmine_gitolite_admin_id_rsa.pub', private_key: '/home/nicolas/.ssh/redmine_gitolite_admin_id_rsa' }
# admin = Gitolite::GitoliteAdmin.new('/home/nicolas/gitolite-admin', settings)

creds = Rugged::Credentials::SshKey.new(username: 'git', publickey: File.expand_path("~/.ssh/redmine_gitolite_admin_id_rsa.pub"), privatekey: File.expand_path("~/.ssh/redmine_gitolite_admin_id_rsa"))

Rugged::Repository.clone_at 'ssh://git@localhost:22/gitolite-admin.git',
                            './gitolite-admin',
                            credentials: creds
```

**Comment by willco007** [comment]
I just applied the fix for #360 to the 1.8.x branch, could you humor me and try that branch? That bug could cause random transport failures which may cause your user auth failure.

**Comment by n-rodriguez** [comment]
> I just applied the fix for #360 to the 1.8.x branch, could you humor me and try that branch?

I've rebuilt libssh2 package on ArchLinux with your branch. Still not working :

```sh
./rugged.rb:30:in `clone_at': Failed to authenticate SSH session: Unable to send userauth-publickey request (Rugged::SshError)
```
